parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -878,11 +880,97 @@ public String getFile() {
     return blocks;
   }
 
-  public void setRequestedSchema(MessageType projection) {
+  private boolean uniqueId(GroupType schema, HashSet<Type.ID> ids) {
```

Comment:
I don't think it is a good practice to modify a set that's passed in. I would expect this to produce a set. If you want to throw an exception because this finds a duplicate ID, then I think it should just throw an exception in this method.